### DISCUSSION TITLE
Pin.PULL_DOWN not available, only Pin.PULL_UP

### DISCUSSION THREAD

**Discussion by rainergo** [micropython/micropython#9884] [Hardware & Peripherals] [answered]
Hi, 
I am using HX711 with ESP8266, this lib (https://github.com/robert-hh/hx711), below's hx711 wirings (see image) and followed this discussion (https://forum.micropython.org/viewtopic.php?f=15&t=5931&start=60):
Code is:
"""
from machine import Pin
from hx711_gpio import HX711
import time

pin_OUT = Pin(12, Pin.IN, pull=Pin.PULL_DOWN)
pin_SCK = Pin(13, Pin.OUT)

hx = HX711(pin_SCK, pin_OUT)

hx.set_gain(128)
time.sleep_ms(50)
hx.tare()

while True:
    if hx.is_ready():
        print(hx.get_value())
    time.sleep(1)
"""
I am getting this error message:
"""
AttributeError: type object 'Pin' has no attribute 'PULL_DOWN'
"""
Changing "Pin.PULL_DOWN" to "Pin.PULL_UP" is working, but then I get negative readings if I put weight on my sensors.
Any suggestions?

![image](https://user-images.githubusercontent.com/55275965/200501407-dac82240-1fb7-4f49-b107-47e147ddf866.png)

**Comment by robert-hh** [comment, marked as answer]
The pull option is not needed, Try to drop it. If it is still needed, add an external resistor pull-down resistor of about 33kOhm. Note, that the polarity of the result also depends on the wiring of the A inputs. If you swap them, you get the opposite polarity.

**Reply by rainergo** [reply]
Hi Robert, 
thanks for your help, it worked (Sry, for late response). 

One more question: What does the function "set_gain()" do and to which value should I set it?

Thanks.

**Reply by robert-hh** [reply]
set_gain() can be used to adapt the output range to the sensitivity of the sensor. If the range of output values is too small, you can e.g. select a higher gain, and the opposite. Note that a gain of 32 requires using channel B of the HX711. It may also be helpful to connect the GND pins at both sides of the board. Sometimes that reduces the noise.

**Reply by rainergo** [reply]
Wow, you are the fastest responder I have ever met. Thanks a lot for your great help. Everthing is fine now.

**Comment by BitByteShark** [comment]
Hi,
I had the same issue trying to use "pull=Pin.PULL_DOWN" with my D1 mini NodeMcu at Pin15(D8) and was able to use "pull=-1" as an alias/workaround. This way you can utilize the builtin pull-down resistor and don't necessarily need the extra resistor.
